src/librustc_resolve/lib.rs

```diff
@@ -179,6 +179,10 @@ pub enum ResolutionError<'a> {
     CannotCaptureDynamicEnvironmentInFnItem,
     /// error E0435: attempt to use a non-constant value in a constant
     AttemptToUseNonConstantValueInConstant,
+    /// error E0437: type is not a member of trait
+    TypeNotMemberOfTrait(Name, &'a str),
+    /// error E0438: const is not a member of trait
+    ConstNotMemberOfTrait(Name, &'a str),
```

Comment:
Nit: I’d group related enum variants together by purpose rather than error code. That is, it would be easier to read the code if `MethodNotMemberOfTrait`, `TypeNotMemberOfTrait` and `ConstNotMemberOfTrait` were all adjacent to each other.